invoice/src/main/java/org/killbill/billing/invoice/generator/FixedAndRecurringInvoiceItemGenerator.java

```diff
@@ -202,21 +204,27 @@ boolean isSameDayAndSameSubscription(final InvoiceItem prevComputedFixedItem, fi
     // Turn a set of events into a list of invoice items. Note that the dates on the invoice items will be rounded (granularity of a day)
     private List<InvoiceItem> processRecurringEvent(final UUID invoiceId, final UUID accountId, final BillingEvent thisEvent, @Nullable final BillingEvent nextEvent,
                                                     final LocalDate targetDate, final Currency currency,
-                                                    final InvoiceItemGeneratorLogger invoiceItemGeneratorLogger, final BillingMode billingMode,
+                                                    final InvoiceItemGeneratorLogger invoiceItemGeneratorLogger,
                                                     final Map<UUID, SubscriptionFutureNotificationDates> perSubscriptionFutureNotificationDate,
                                                     final InternalCallContext internalCallContext) throws InvoiceApiException {
 
         try {
             final List<InvoiceItem> items = new ArrayList<InvoiceItem>();
 
-            // For FIXEDTERM phases we need to stop when the specified duration has been reached
-            final LocalDate maxEndDate = thisEvent.getPlanPhase().getPhaseType() == PhaseType.FIXEDTERM ?
-                                         thisEvent.getPlanPhase().getDuration().addToLocalDate(internalCallContext.toLocalDate(thisEvent.getEffectiveDate())) :
-                                         null;
 
             // Handle recurring items
             final BillingPeriod billingPeriod = thisEvent.getBillingPeriod();
             if (billingPeriod != BillingPeriod.NO_BILLING_PERIOD) {
+
+                final Plan currentPlan = thisEvent.getPlan();
+                Preconditions.checkNotNull(currentPlan, String.format("Unexpected null Plan name event = %s", thisEvent));
```

Comment:
No need for `String.format` with `Preconditions`.

Reply:
ack :-)